src/routes/clusters.py

```diff
@@ -0,0 +1,86 @@
+"""
+Routes for Spark cluster management.
+"""
+
+import logging
+import math
+
+from fastapi import APIRouter, Depends, status
+
+from src.service import kb_auth
+from src.service.dependencies import auth
+from src.service.exceptions import ConfigurationLimitExceededError
+from src.service.kb_auth import AdminPermission
+from src.service.models import (
+    DEFAULT_MASTER_MEMORY,
+    DEFAULT_WORKER_MEMORY,
+    SparkClusterConfig,
+    SparkClusterCreateResponse,
+)
+from src.spark_manager import KubeSparkManager
+
+logger = logging.getLogger(__name__)
+
+# Create a router for cluster endpoints
+router = APIRouter(prefix="/clusters", tags=["clusters"])
+
+
+@router.post(
+    "",
+    response_model=SparkClusterCreateResponse,
+    status_code=status.HTTP_201_CREATED,
+    summary="Create a new Spark cluster",
+    description="Creates a new Spark cluster for the authenticated user with the specified configuration.",
+)
+async def create_cluster(
+    config: SparkClusterConfig, user: kb_auth.KBaseUser = Depends(auth)
+) -> SparkClusterCreateResponse:
+    """Create a new Spark cluster for the authenticated user."""
+
+    manager = KubeSparkManager(
+        username=str(user.user),
+    )
+
+    # Validate config against defaults for non-admin users
+    if user.admin_perm != AdminPermission.FULL:
+        default_config = SparkClusterConfig(
+            worker_memory=DEFAULT_WORKER_MEMORY,  # type: ignore[assignment]
+            master_memory=DEFAULT_MASTER_MEMORY,  # type: ignore[assignment]
+        )
+
+        exceeds_limits = (
+            config.worker_count > default_config.worker_count
+            or config.worker_cores > default_config.worker_cores
+            or config.worker_memory > default_config.worker_memory
+            or config.master_cores > default_config.master_cores
+            or config.master_memory > default_config.master_memory
+        )
+
+        if exceeds_limits:
+            raise ConfigurationLimitExceededError(
+                "Configuration exceeds default limits for non-admin users. "
+                f"Max Workers: {default_config.worker_count}, "
+                f"Max Worker Cores: {default_config.worker_cores}, "
+                f"Max Worker Memory: {default_config.worker_memory.human_readable()}, "
+                f"Max Master Cores: {default_config.master_cores}, "
+                f"Max Master Memory: {default_config.master_memory.human_readable()}."
+            )
```

Comment:
It'd be more helpful if it told the user exactly which limit was exceeded but this is fine